miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/MisoWebUtils.java

```diff
@@ -78,14 +78,31 @@ public static String generateErrorDivMessage(String errorMessage, String excepti
     return checks;
   }
 
+  /**
+   * Similar to checkDirectory, but for single files.
+   * 
+   * @param path of type File
+   * @return boolean true if the file exists, false if not
+   * @throws IOException when the file doesn't exist
+   */
+  private static boolean checkFile(File path) throws IOException {
+    boolean storageOk = path.exists();
+    if (!storageOk) {
+      StringBuilder sb = new StringBuilder("The file [" + path.toString() + "] doesn't exist.");
+      throw new IOException(sb.toString());
+    } else {
+      log.info("File (" + path + ") OK.");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

Reply:
Null parent is handled for plain samples. For detailed samples, it would make sense for this method to return true if `child` is an Identity and false otherwise